library/oneview_server_profile.py

```diff
@@ -525,9 +523,20 @@ def is_virtual_or_physical(defined_type):
                         volume.pop(SPKeys.LUN, None)
 
     def __get_available_server_hardware_uri(self):
-        scope_uris = self.data.get('initialScopeUris', [])
-        scope_uri = '%20OR%20'.join(scope_uris)
 
+        # Retrive scopeUri from oneview for scoped user if initialScopeUris is null
+        scope_uri = ''
+        if self.data.get('initialScopeUris'):
+            scope_uris = self.data.get('initialScopeUris', [])
+            scope_uri = '%20OR%20'.join(scope_uris)
+        else:
+            user_name = self.oneview_client._OneViewClient__connection._cred['userName']
+            scope_user = self.oneview_client.users.get_by_userName(user_name)
+            if scope_user and user_name != 'Administrator':
+                permissions = scope_user.data["permissions"]
+                scope_uris = set([each_role.get("scopeUri") for each_role in permissions])
+                scope_uri = '%20OR%20'.join(list(scope_uris))
```

Comment:
why is this required when you are not using scope_uri anywhere..?

Reply:
its being used in line numbers  557 and 567.

Reply:
ok